youtube_dl/extractor/xvideos.py

```diff
@@ -45,7 +99,16 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
 
         webpage = self._download_webpage(
-            'https://www.xvideos.com/video%s/' % video_id, video_id)
+            'https://www.xvideos.com/video%s/' % video_id, video_id, expected_status=404)
+
+        status_404 = get_element_by_class("status-404", webpage)
+        if status_404:
+            reg_not_found = r'<div[^>]+id=["\']content["\']>[\r\n]*?<h1[^>]*>(?P<reason>[^<]*)'
+            deleted = get_element_by_class("text-danger", status_404)
+            not_found = self._search_regex(reg_not_found, status_404, 'reason', default=None, group='reason')
+            reason = deleted or not_found or None
+            if reason:
```

Comment:
If it's not found you will try extracting from 404 page. Remove all this 404 mess.